maro/simulator/scenarios/finance/common/trader.py

```diff
@@ -0,0 +1,252 @@
+                    deal_money = deal_price * deal_volume
+                    if deal_money > remaining_money:
+                        deal_volume -= math.ceil((deal_money - remaining_money) / deal_price)
+
+                    # if min buy unit is configed
+                    if TradeConstrain.min_buy_unit.value in self._trade_constrain.keys() \
+                            and self._trade_constrain[TradeConstrain.min_buy_unit.value] != 0:
+                        odd_volume = deal_volume % self._trade_constrain[TradeConstrain.min_buy_unit.value]
+                        if odd_volume != 0:
+                            deal_volume -= odd_volume
+
+                else:  # sell
+                    avalib_volume = cur_data[order_action.item_index].account_hold_num
+                    if avalib_volume < deal_volume:
+                        deal_volume = avalib_volume
+
+                    # if min sell unit is configed
+                    if TradeConstrain.min_sell_unit.value in self._trade_constrain.keys() \
+                            and self._trade_constrain[TradeConstrain.min_sell_unit.value] != 0:
+                        odd_volume = abs(deal_volume) % self._trade_constrain[TradeConstrain.min_sell_unit.value]
+                        if odd_volume != 0:
+                            if odd_volume != avalib_volume % self._trade_constrain[TradeConstrain.min_sell_unit.value]:
+                                deal_volume -= odd_volume
+
+                for commission_handler in self._commission_handlers:
+                    if is_buy:
+                        commission_charge += commission_handler.execute(deal_price, deal_volume)
+                    else:
+                        commission_charge += commission_handler.execute(deal_price, -deal_volume)
+
+                commission_charge = int(commission_charge * 100) / 100
+
+                # if remaining_money is not enough for commission
+                # TODO: sell out money not cover commission not implimented
+                if is_buy:
+                    expected_remaining = remaining_money - deal_volume * deal_price - commission_charge
+                    if expected_remaining < 0:
+                        deal_volume -= math.ceil((-expected_remaining) / deal_price)
+
+                        if TradeConstrain.min_buy_unit.value in self._trade_constrain.keys() \
+                                and self._trade_constrain[TradeConstrain.min_buy_unit.value] != 0:
+                            odd_volume = deal_volume % self._trade_constrain[TradeConstrain.min_buy_unit.value]
+                            if odd_volume != 0:
+                                deal_volume -= odd_volume
+
+                        if deal_volume < 0:
+                            deal_volume = 0
+                            is_success = False
+
+                        # recalculate commission charge
+                        commission_charge = 0
```

Comment:
Please give a proposal of the calculation of slippage and deal volume. We can have further discussion before implementation.